introspection-engine/connectors/sql-introspection-connector/src/schema_describer_loading.rs

```diff
@@ -22,6 +22,19 @@ pub async fn load_describer<'a>(
                 if provider == Some(POSTGRES.provider_name()) {
                     circumstances |= Circumstances::CockroachWithPostgresNativeTypes;
                 }
+            } else {
+                let pgversion_result = connection
+                    .query_raw("select current_setting('server_version_num')::integer as version;", &[])
+                    .await?;
+                let pgversion = pgversion_result
+                    .get(0)
+                    .and_then(|r| r.get("version"))
+                    .and_then(|v| v.as_i32());
+
+                match pgversion {
+                    Some(version) if version >= 100000 => circumstances |= Circumstances::CanPartitionTables,
+                    _ => (),
+                }
```

Comment:
:+1:

Reply:
Just a note that we'll need to move that logic to sql-migration-connector when the introspection tests move there.

Reply:
Yep, but for now this is the one that's used. I didn't want to per-emptively change code that's not actually used in this case.

Reply:
It's used in production though. This code path is only used in tests. The introspection engine is not behind introspection in Prisma CLI anymore.